src/java/org/apache/cassandra/db/guardrails/Values.java

```diff
@@ -0,0 +1,155 @@
+     * @param ignoreAction An action called on the subset of {@code values} that should be ignored. This action
+     *                     should do whatever is necessary to make sure the value is ignored.
+     * @param state        The client state, used to skip the check if the query is internal or is done by a superuser.
+     *                     A {@code null} value means that the check should be done regardless of the query.
+     */
+    public void guard(T value, Consumer<T> ignoreAction, @Nullable ClientState state)
+    {
+        if (!enabled(state))
+            return;
+
+        Config<T> config = configProvider.apply(state);
+
+        Set<T> ignored = config.getIgnored();
+        if (ignored.contains(value))
+        {
+            warn(format("Ignoring provided value %s as it is not supported for %s (ignored values are: %s)",
+                        value, what, ignored.toString()));
+            ignoreAction.accept(value);
+        }
+
+        Set<T> disallowed = config.getDisallowed();
+        if (disallowed.contains(value))
+            abort(format("Provided value %s is not allowed for %s (disallowed values are: %s)",
+                         value, what, disallowed.toString()));
+    }
+
+    /**
+     * Triggers a warning for each of the provided values that are disallowed by this guardrail and triggers an action
+     * to ignore them. If the values are disallowed it will abort the operation.
+     *
+     * @param values       The values to check.
+     * @param ignoreAction An action called on the subset of {@code values} that should be ignored. This action
+     *                     should do whatever is necessary to make sure the value is ignored.
+     * @param state        The client state, used to skip the check if the query is internal or is done by a superuser.
+     *                     A {@code null} value means that the check should be done regardless of the query.
+     */
+    public void guard(Set<T> values, Consumer<T> ignoreAction, @Nullable ClientState state)
+    {
+        if (!enabled(state))
+            return;
+
+        Config<T> config = configProvider.apply(state);
+
+        Set<T> ignored = config.getIgnored();
+        Set<T> toIgnore = Sets.intersection(values, ignored);
+        if (!toIgnore.isEmpty())
+        {
+            warn(format("Ignoring provided values %s as they are not supported for %s (ignored values are: %s)",
+                        toIgnore.stream().sorted().collect(Collectors.toList()), what, ignored.toString()));
+            toIgnore.forEach(ignoreAction);
+        }
+
+        Set<T> disallowed = config.getDisallowed();
+        Set<T> toDisallow = Sets.intersection(values, disallowed);
+        if (!toDisallow.isEmpty())
+            abort(format("Provided values %s are not allowed for %s (disallowed values are: %s)",
+                         toDisallow.stream().sorted().collect(Collectors.toList()), what, disallowed.toString()));
```

Comment:
can we do disallow before ignore?  ignore mutates, so it could be a problem in the future if we mutate then abort

Reply:
Good point, done.